### PR TITLE
UIDATIMP-1217: Create the field mapping profile for Orders and Order Lines: View

### PR DESCRIPTION
**Pull request by OleksandrHladchenko1**
## Purpose

* To display the details of the Order field mapping profile that was created (same general format as all other field mapping profiles)

## Approach

* Created `<MappingOrderDetails />` component to display the information
* Added `getFieldValueByPath` method to get `value` from the field, using `path` (when `name` fields are equal)

## Refs

[UIDATIMP-1217](https://issues.folio.org/browse/UIDATIMP-1217)

## Screenshots

https://user-images.githubusercontent.com/85172747/198010294-7f31db25-7031-4348-8b96-97080746caf8.mp4

### CODE REVIEW

**Comment by OleksandrHladchenko1** [comment]
@folio-org/fe-tl-reviewers please, take a look

File: src/settings/MappingProfiles/detailsSections/view/MappingOrderDetails.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+
+import { Accordion, AccordionSet } from '@folio/stripes/components';
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/MappingOrderDetails.js:5]:
```suggestion
import {
Accordion,
AccordionSet,
} from '@folio/stripes/components';
```

File: src/settings/MappingProfiles/detailsSections/view/MappingOrderDetails.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import { OrderInformation } from './OrderDetailsSection/OrderInformation';
+import { OrderDetails } from './OrderDetailsSection/OrderDetails';
+import { POLineDetails } from './OrderDetailsSection/POLineDetails';
+import { Vendor } from './OrderDetailsSection/Vendor';
+import { CostDetails } from './OrderDetailsSection/CostDetails';
+import { FundDistribution } from './OrderDetailsSection/FundDistribution';
+import { Location } from './OrderDetailsSection/Location';
+import { PhysicalResourceDetails } from './OrderDetailsSection/PhysicalResourceDetails';
+import { EResourcesDetails } from './OrderDetailsSection/EResourcesDetails';
```
Comment (mariia-aloshyna) [inline src/settings/MappingProfiles/detailsSections/view/MappingOrderDetails.js:15]: Export components in `index.js` and import all from `./OrderDetailsSection`;

File: src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/EResourcesDetails.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  const userLimit = getFieldValue(mappingDetails, 'userLimit', 'value');
+  const resourceUrl = getFieldValue(mappingDetails, 'resourceUrl', 'value');
+
+  const activatedCheckbox = renderCheckbox('order.eResourcesDetails.activated', activated);
+  const trialCheckbox = renderCheckbox('order.eResourcesDetails.trial', trial);
+
+  const formattedActivationDue = (
+    <FormattedDate
+      value={activationDue}
+      day="2-digit"
```
Comment (Dmitriy-Litvinenko) [inline src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/EResourcesDetails.js:40]: Could you please investigate possible to move `day="2-digit" month="2-digit" year="numeric"` to variable and reuse it?

File: src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/FundDistribution.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+} from '../../utils';
+import { TRANSLATION_ID_PREFIX } from '../../constants';
+import { mappingProfileFieldShape } from '../../../../../utils';
+
+export const FundDistribution = ({ mappingDetails }) => {
+  const noValueElement = <NoValue />;
+
+  const fundDistributions = getFieldValue(mappingDetails, 'fundDistribution', 'subfields');
+
+  const fundDistributionsVisibleColumns = ['fundId', 'expenseClassId', 'value'];
```
Comment (Dmitriy-Litvinenko) [inline src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/FundDistribution.js:27]: Could you please investigate possible to move  `'fundId', 'expenseClassId', 'value' `to const and reuse it in this case and in similar cases across all pull?

File: src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/FundDistribution.js [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    expenseClassId: (
+      <FormattedMessage id={`${TRANSLATION_ID_PREFIX}.order.fundDistributions.expenseClassId`} />
+    ),
+    value: (
+      <FormattedMessage id={`${TRANSLATION_ID_PREFIX}.order.fundDistributions.value`} />
+    ),
+  };
+
+  const fundDistributionsFormatter = {
+    fundId: x => x?.fundId || noValueElement,
```
Comment (Dmitriy-Litvinenko) [inline src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/FundDistribution.js:42]: Could you please investigate possible to rename `x`  in this case and in similar cases across all pull to make the code more readable? Just for example `distribution` for current case

File: src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/OrderDetails.js [hunk 6]
```diff
@@ -0,0 +1,256 @@
...
+      year="numeric"
+    />
+  );
+  const formattedSubscriptionTo = (
+    <FormattedDate
+      value={subscriptionTo}
+      day="2-digit"
+      month="2-digit"
+      year="numeric"
+      format="YYYY-MM-DD"
```
Comment (Dmitriy-Litvinenko) [inline src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/OrderDetails.js:113]: Could you please investigate can we create const `DATE_FORMAT = 'YYYY-MM-DD'; ` and reuse it across all project?

**Review by mariia-aloshyna — APPROVED** [review]

File: src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/EResourcesDetails.js [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+  const userLimit = getFieldValue(mappingDetails, 'userLimit', 'value');
+  const resourceUrl = getFieldValue(mappingDetails, 'resourceUrl', 'value');
+
+  const activatedCheckbox = renderCheckbox('order.eResourcesDetails.activated', activated);
+  const trialCheckbox = renderCheckbox('order.eResourcesDetails.trial', trial);
+
+  const formattedActivationDue = (
+    <FormattedDate
+      value={activationDue}
+      day={TWO_DIGIT}
```
Comment (Dmitriy-Litvinenko) [inline src/settings/MappingProfiles/detailsSections/view/OrderDetailsSection/EResourcesDetails.js:48]: Could you please investigate should we move this 3 line to 1 const and reuse it? Example below.
const BASE_FORMATTED_DATE = {
  day: TWO_DIGIT,
  month: TWO_DIGIT,
  year: NUMERIC,
};

**Review by Dmitriy-Litvinenko — APPROVED** [review]

**Pull request merged**